From d9071bf700a72ce25fcacce687d34cf7bcfd2546 Mon Sep 17 00:00:00 2001
From: Tom Salata <toms@tpi.com>
Date: Mon, 18 Sep 2023 10:47:15 -0700
Subject: [PATCH 6/6] dts: g2h-solo-11f: Use Focal Tech touchscreen driver

Use drivers/input/touchscreen/focaltech_touch for the new Focal Tech
FT7811 touch controller in place of the evervision driver.

Signed-off-by: Tom Salata <toms@tpi.com>
---
 arch/arm/boot/dts/imx6dl-g2h-11f.dts | 18 ++++++++++++++++--
 1 file changed, 16 insertions(+), 2 deletions(-)

diff --git a/arch/arm/boot/dts/imx6dl-g2h-11f.dts b/arch/arm/boot/dts/imx6dl-g2h-11f.dts
index 2baad782ec71..778d5a096f69 100644
--- a/arch/arm/boot/dts/imx6dl-g2h-11f.dts
+++ b/arch/arm/boot/dts/imx6dl-g2h-11f.dts
@@ -11,6 +11,7 @@
 /dts-v1/;
 
 #include <dt-bindings/interrupt-controller/irq.h>
+#include <dt-bindings/input/input.h>
 #include "imx6dl.dtsi"
 #include "imx6dl-g2h.dtsi"
 
@@ -67,8 +68,21 @@
 	status = "disabled";
 };
 
-&evervision_ts {
-    status = "okay";
+&i2c1 {
+	focaltech_ts: fts_ts@38 {
+		compatible = "focaltech,fts";
+		reg = <0x38>;
+		interrupt-parent = <&gpio4>;
+		interrupts = <5 0>;
+		focaltech,panel-type = <FT7811>;
+		focaltech,irq-gpio = <&gpio4 5 0x0>;
+		focaltech,reset-gpio = <&gpio7 8 0>;
+		focaltech,max-touch-number = <1>;
+		focaltech,display-coords = <0 0 2047 2047>;
+		focaltech,swap-x-y;
+		focaltech,single-touch;
+		status = "okay";
+	};
 };
 
 &spiflash {
-- 
2.25.1

